### PR TITLE
Feature/sg 647 checkpoints tutorial

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

File: documentation/source/Checkpoints.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+class ModelB(torch.nn.Module):
+    def __init__(self):
+        super(ModelB, self).__init__()
+        self.conv1 = torch.nn.Conv2d(3, 6, 5)
+        self.CONV2 = torch.nn.Sequential([torch.nn.Conv2d(6, 16, 5)])
+        
+```
+
+Notice the above networks have identical weight structures but will have different keys in their `state_dict`s.
+This is why loading a checkpoint from either one to the other, using `strict=True`, will fail and crash. Using `strict=True` will not crash and successfully load the first layer's weights only.
```
Comment (ofrimasad) [inline documentation/source/Checkpoints.md:175]: the second one should be `strict=false`

File: documentation/source/Checkpoints.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+
+```shell
+> python train_from_recipe.py --config-name=cifar10_resnet experiment_name=cifar_experiment training_hyperparams.resume=True training_hyperparams.max_epochs=300
+...
+...
+> python train_from_recipe.py --config-name=cifar10_resnet experiment_name=cifar_experiment training_hyperparams.resume=True training_hyperparams.max_epochs=400
+```
+
+This means that such flexibility comes with its price: we must be aware of any change in parameters (by command line overrides or hard-coded changes inside the yaml file configurations) if we wish to resume training.
```
Comment (ofrimasad) [inline documentation/source/Checkpoints.md:267]: please rephrase.
"However, this flexibility comes with a price"

**Review by ofrimasad — CHANGES_REQUESTED** [review]
very nice. 
some comments regarding phrasing and formatting are inline

File: documentation/source/Checkpoints.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    When a model is training, the performance changes as it sees more data. It is a best practice to save the state of a model throughout the training process. This gives you a version of the model, a checkpoint, at each key point during the development of the model. Once training has been completed, use the checkpoint corresponding to the best performance you found during the training process.
+    
+    Checkpoints also enable your training to resume where it was in case the training process is interrupted.
+
+## Checkpoints Saving in SG: Which, When, and Where?
+
+From the previous subsection, we understand that checkpoints saved at different times during training have different purposes.
+That's why in SG, multiple checkpoints are saved throughout training:
+
+| Checkpoint Filename        | When is it saved?           |
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Checkpoints.md]: fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422

File: documentation/source/Checkpoints.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+| `average_model.pth` | At the end of training - composed of 10 best models according to  [metric_to_watch](https://github.com/Deci-AI/super-gradients/blob/69d8d19813964022af192a34b6e7853edac34a75/src/super_gradients/recipes/training_hyperparams/default_train_params.yaml#L39) and will only be save when the training_param `average_best_models`=True.     |
+| `ckpt_epoch_{EPOCH_INDEX}.pth`| At the end of a fixed epoch number `EPOCH_INDEX` if it is specified through `save_ckpt_epoch_list` training_param |
+
+#### Where are the checkpoint files saved?
+
+The checkpoint files will be saved at <PATH_TO_CKPT_ROOT_DIR>/experiment_name/.
+
+The user controls the checkpoint root directory, which can be passed to the `Trainer` constructor through the `ckpt_root_dir` argument.
+
+When working with a cloned version of SG, one can leave out the `ckpt_root_dir` arg, and checkpoints will be saved in the `checkpoints` module.
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Checkpoints.md]: fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422

File: documentation/source/Checkpoints.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+## Checkpoint Structure in SG
+
+Checkpoints in SG are instances of [state_dict](https://pytorch.org/tutorials/beginner/saving_loading_models.html#what-is-a-state-dict).
+They hold additional information about the model's training besides the model weights.
+
+The checkpoint keys:
+
+"net": The network's state_dict (state_dict).
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Checkpoints.md]: fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422

File: documentation/source/Checkpoints.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+Suppose we wish to load the weights from `ckpt_best.pth`. We can simply pass its path to the `checkpoint_path` argument in `models.get(...)`:
+
+   ```python
+        from super_gradients.training import models
+        from super_gradients.common.object_names import Models
+        
+        model = models.get(model_name=Models.RESNET18, num_classes=10, checkpoint_path="/path/to/my_checkpoints_folder/my_resnet18_training_experiment/ckpt_best.pth")
+   ```
+
+`Important: when loading SG-trained checkpoints using models.get(...), if the network was trained with EMA, the EMA weights will be the ones loaded. `
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply documentation/source/Checkpoints.md]: fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422

File: documentation/source/Checkpoints.md (hunk 1 above)
Comment (shaydeci) [reply documentation/source/Checkpoints.md]: fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422

File: documentation/source/Checkpoints.md (hunk 2 above)
Comment (shaydeci) [reply documentation/source/Checkpoints.md]: fixed https://github.com/Deci-AI/super-gradients/pull/703/commits/ad2036e85f0b857fe9123ca0407553f68a433422

**Review by ofrimasad — APPROVED** [review]
LGTM
